### PR TITLE
e2e test

### PR DESCRIPTION
**Pull request by apoorvsadana**

### CODE REVIEW

File: src/tests/server.rs [hunk 1]
```diff
@@ -1,16 +1,115 @@
...
 
+async fn send_create_indexer_request(
+    client: Client<HttpConnector>,
+    script_path: &str,
+    addr: SocketAddr,
+) -> Response<Body> {
+    let mut mpart = MultipartRequest::default();
+
+    mpart.add_file("script.js", script_path);
+    mpart.add_field("webhook_url", "https://webhook.site/bc2ca42e-a8b2-43cf-b95c-779fb1a6bbbb");
```
Comment (EvolveArt) [inline src/tests/server.rs:33]: add webhook_url as param or const

File: src/tests/server.rs [hunk 2]
```diff
@@ -1,16 +1,115 @@
...
+                .body(Body::empty())
+                .unwrap(),
+        )
+        .await
+        .unwrap();
+    // assert the response of the request
+    assert_eq!(response.status(), StatusCode::OK);
+}
+
+async fn assert_queue_contains_message_with_indexer_id(queue_url: &str, indexer_id: Uuid) {
```
Comment (EvolveArt) [inline src/tests/server.rs:82]: move all these functions to utils

Comment (EvolveArt) [reply src/tests/server.rs:82]: and add some basic rustdoc

File: src/tests/server.rs [hunk 3]
```diff
@@ -69,31 +168,175 @@ async fn create_indexer(#[future] setup_server: SocketAddr) {
...
-                .uri(format!("http://{}/v1/indexers", addr))
-                .body(Body::wrap_stream(mpart))
-                .unwrap(),
-        )
-        .await
-        .unwrap();
-
-    assert_eq!(response.status(), StatusCode::OK);
+    // Create indexer
+    let response = send_create_indexer_request(client.clone(), "./src/tests/scripts/test.js", addr).await;
```
Comment (EvolveArt) [inline src/tests/server.rs:177]: add the path as a variable

File: src/tests/server.rs [hunk 4]
```diff
@@ -69,31 +168,175 @@ async fn create_indexer(#[future] setup_server: SocketAddr) {
...
 
     let body = hyper::body::to_bytes(response.into_body()).await.unwrap();
     let body: IndexerModel = serde_json::from_slice(&body).unwrap();
 
     assert_eq!(body.status, IndexerStatus::Created);
     assert_eq!(body.indexer_type, IndexerType::Webhook);
     assert_eq!(body.target_url, "https://webhook.site/bc2ca42e-a8b2-43cf-b95c-779fb1a6bbbb");
+
+    // check if the file exists on S3
+    assert!(
```
Comment (EvolveArt) [inline src/tests/server.rs:187]: assert_s3_exists!

**Review by EvolveArt — COMMENTED** [review]
Looks good! Could we make this more generic so we can re-use the code when we add Postgres sink support ?

File: src/tests/server.rs (hunk 1 above)
Comment (apoorvsadana) [reply src/tests/server.rs]: fixed

File: src/tests/server.rs (hunk 2 above)
Comment (apoorvsadana) [reply src/tests/server.rs]: fixed

File: src/tests/server.rs (hunk 3 above)
Comment (apoorvsadana) [reply src/tests/server.rs]: fixed

File: src/tests/server.rs (hunk 4 above)
Comment (apoorvsadana) [reply src/tests/server.rs]: fixed

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**